includes/multi-currency/Analytics.php

```diff
@@ -291,17 +292,26 @@ public function filter_join_clauses( array $clauses, $context ): array {
 		$context_page  = $context_parts[0] ?? 'generic';
 
 		$prefix                   = 'wcpay_multicurrency_';
-		$currency_tbl             = $prefix . 'currency_postmeta';
-		$default_currency_tbl     = $prefix . 'default_currency_postmeta';
-		$exchange_rate_tbl        = $prefix . 'exchange_rate_postmeta';
-		$stripe_exchange_rate_tbl = $prefix . 'stripe_exchange_rate_postmeta';
+		$currency_tbl             = $prefix . 'currency_meta';
+		$default_currency_tbl     = $prefix . 'default_currency_meta';
+		$exchange_rate_tbl        = $prefix . 'exchange_rate_meta';
+		$stripe_exchange_rate_tbl = $prefix . 'stripe_exchange_rate_meta';
+
+		// Allow this to work with custom order tables as well.
+		if ( class_exists( OrderUtil::class ) && OrderUtil::custom_orders_table_usage_is_enabled() ) {
+			$meta_table = $wpdb->prefix . 'wc_orders_meta';
+			$id_field   = 'order_id';
+		} else {
+			$meta_table = $wpdb->postmeta;
+			$id_field   = 'post_id';
+		}
 
 		// If this is a supported context, add the joins. If this is an unsupported context, see if we can add the joins.
 		if ( $this->is_supported_context( $context ) && ( in_array( $context_page, self::SUPPORTED_CONTEXTS, true ) || $this->is_order_stats_table_used_in_clauses( $clauses ) ) ) {
-			$clauses[] = "LEFT JOIN {$wpdb->postmeta} {$currency_tbl} ON {$wpdb->prefix}wc_order_stats.order_id = {$currency_tbl}.post_id AND {$currency_tbl}.meta_key = '_order_currency'";
-			$clauses[] = "LEFT JOIN {$wpdb->postmeta} {$default_currency_tbl} ON {$wpdb->prefix}wc_order_stats.order_id = {$default_currency_tbl}.post_id AND ${default_currency_tbl}.meta_key = '_wcpay_multi_currency_order_default_currency'";
-			$clauses[] = "LEFT JOIN {$wpdb->postmeta} {$exchange_rate_tbl} ON {$wpdb->prefix}wc_order_stats.order_id = {$exchange_rate_tbl}.post_id AND ${exchange_rate_tbl}.meta_key = '_wcpay_multi_currency_order_exchange_rate'";
-			$clauses[] = "LEFT JOIN {$wpdb->postmeta} {$stripe_exchange_rate_tbl} ON {$wpdb->prefix}wc_order_stats.order_id = {$stripe_exchange_rate_tbl}.post_id AND ${stripe_exchange_rate_tbl}.meta_key = '_wcpay_multi_currency_stripe_exchange_rate'";
+			$clauses[] = "LEFT JOIN {$meta_table} {$currency_tbl} ON {$wpdb->prefix}wc_order_stats.order_id = {$currency_tbl}.{$id_field} AND {$currency_tbl}.meta_key = '_order_currency'";
```

Comment:
This clause will need to be updated because we no longer store `_order_currency` as a meta key in the new COT meta tables. Instead, it is stored in the `wp_wc_orders` table under the key `currency`.

I think we should do something like this:

- If COT is not enabled, use the query as it is currently
- If using COT, instead of joining on `currency_tbl`, do a `LEFT JOIN` on `wp_wc_orders` (joining on the order ID), then update `filter_select_clauses` function to get the `currency` value from this table instead of getting it from the meta table as it does currently.

This value populates the `Customer Currency` field which you can see in `Analytics > Orders`

Reply:
Great catch, @dmallory42! 🙌 

I changed the column in https://github.com/Automattic/woocommerce-payments/pull/4677/commits/d59a0d83790da30a5160d4b80609609d1550684f to address this.

As an additional step, I deleted a couple of orders from `wp_posts` and `wp_postmeta` to make sure that they still appear in Analytics, sourced from the orders table. It worked great! 🎊